src/main/java/seedu/agendum/logic/parser/Parser.java

```diff
@@ -95,16 +98,35 @@ public Command parseCommand(String userInput) {
      * @return the prepared command
      */
     private Command prepareAdd(String args){
-        final Matcher matcher = TASK_DATA_ARGS_FORMAT.matcher(args.trim());
+        final Matcher matcher = ADD_ARGS_FORMAT.matcher(args.trim());
         // Validate arg string format
-        if (!matcher.matches()) {
+        if (!matcher.find()) {
             return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
         }
+
         try {
-            return new AddCommand(
-                    matcher.group("name"),
-                    getTagsFromArgs(matcher.group("tagArguments"))
-            );
+            String name = matcher.group("name");
+            Optional<LocalDateTime> byDateTime = DateTimeParser.parseString("");
+            Optional<LocalDateTime> fromDateTime = DateTimeParser.parseString(matcher.group("fromArg"));
+            Optional<LocalDateTime> toDateTime = DateTimeParser.parseString(matcher.group("toArg"));
+
+            if (byDateTime.isPresent()) {
+                return new AddCommand(
+                        name,
+                        byDateTime,
+                        getTagsFromArgs("test")
```

Comment:
Temporary to placate the constructor. An upcoming PR that removes tagging support will clean this up